wezterm-gui/src/termwindow/charselect.rs

```diff
@@ -268,45 +269,57 @@ fn compute_matches(selection: &str, aliases: &[Alias], group: CharSelectGroup) -
+                Character::Emoji(d) => d.as_str().chars().next().unwrap(),
+            };
+            let alias_result = matcher
+                .fuzzy_match(&entry.name, selection)
+                .map(|score| MatchResult::new(row_idx, score, selection, aliases));
+            if let Some(value) = match &numeric_selection {
+                Some(sel) => {
+                    let codepoints = entry.codepoints();
+                    if codepoints == *sel {
+                        Some(MatchResult {
+                            row_idx,
+                            score: i64::max_value(),
+                        })
+                    } else {
+                        let number_result = matcher
+                            .fuzzy_match(&codepoints, &sel)
+                            .map(|score| MatchResult::new(row_idx, score, sel, aliases));
+
+                        match (alias_result, number_result) {
+                            (
+                                Some(MatchResult { score: a, .. }),
+                                Some(MatchResult { score: b, .. }),
+                            ) => Some(MatchResult {
                                 row_idx,
-                                score: i64::max_value(),
-                            })
-                        } else {
-                            let number_result = matcher
-                                .fuzzy_match(&codepoints, &sel)
-                                .map(|score| MatchResult::new(row_idx, score, sel, aliases));
-
-                            match (alias_result, number_result) {
-                                (
-                                    Some(MatchResult { score: a, .. }),
-                                    Some(MatchResult { score: b, .. }),
-                                ) => Some(MatchResult {
-                                    row_idx,
-                                    score: a.max(b),
-                                }),
-                                (Some(a), None) | (None, Some(a)) => Some(a),
-                                (None, None) => None,
-                            }
+                                score: a.max(b),
+                            }),
+                            (Some(a), None) | (None, Some(a)) => Some(a),
+                            (None, None) => None,
                         }
                     }
-                    None => alias_result,
                 }
-            })
-            .collect();
+                None => alias_result,
+            } {
+                if let Some(other) = matches.get(&character) {
+                    // if the character was inserted already, replace it only if the score is improved
+                    if other.score < value.score {
+                        matches.insert(character, value);
+                    }
+                } else {
+                    matches.insert(character, value);
+                }
```

Comment:
Consider using something like this:

```rust
let existing = matches.entry(glyph).or_insert(value);
// Replace an existing entry only if the score improved
if value.score > existing.score {
   *existing = value;
}
```

Reply:
This isn't really possible, `or_insert()` will move `value` so you can't use it in the line below that.

The most elegant way to coalesce the two insert statements would be to use some negated `is_some_and()` `let` statement, but this isn't a stable Rust feature in wezterm's minimal Rust requirement.